python/lsst/ts/love/csc/authorization_model.py

```diff
@@ -0,0 +1,180 @@
+# This file is part of ts_love_csc.
+#
+# Developed for Vera C. Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import logging
+
+from .utils import parse_auth_request
+
+
+class AuthorizationModel:
+    """Handle CSC authorization.
+
+    Attributes
+    ----------
+    authorization : `dict`
+        Dictionary with the current state of the system auth list.
+    """
+
+    def __init__(self, log=None) -> None:
+
+        self.log = (
+            logging.getLogger(type(self).__name__)
+            if log is None
+            else log.getChild(type(self).__name__)
+        )
+        self.authorization = dict()
+        self._auto_accept_request = True
+
+    def get_authorization(self, component, authorized_users, non_authorized_cscs):
+        """Get the result of authorizing users/cscs to control a component.
+
+        This method returns what would be the result of applying a certain
+        authorization to a component in the list. It does not implement the
+        authorization. To apply it, use set_authorization with the return
+        value of this method.
+
+        Parameters
+        ----------
+        component : `str`
+            Name of the component to authorize.
+        authorized_users : `str`
+            Comma seperated list of users to authorize.
+        non_authorized_cscs : `str`
+            Comma seperated list of cscs to unauthorize.
+
+        Returns
+        -------
+        component_authorization : `dict`
+            Dictionary with the result of applying the authorization.
```

Comment:
I strongly suggest returning a class or dataclass instead of a dict. A dataclass is fine if you are just returning a struct with no behavior. A regular class is better if you wish to add behavior to the class (which you might very well want to do).

The advantages to either vs a dict:
* This provides a place to document the fields.
* It avoids misspellings.
* It makes the code a bit easier to read (`.foo` instead of `["foo"]`).